Automattic/jetpack
# Update utils to typescript

## Proposed changes:

* Update the utils functions to use Typescript

### Other information:

- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated commeP2nt below with a script to run)?

## Jetpack product discussion

P2: p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Go to `/wp-admin/admin.php?page=my-jetpack#/` and make sure the "Your plans" section works as expected (the utils functions are essentially only used here, outside of one being used on the golden token page)
![image](https://github.com/Automattic/jetpack/assets/65001528/af40c5d4-28df-4c97-94f7-d63754072a3a)
